doc/dev/transforms.rst

```diff
@@ -0,0 +1,594 @@
+            SymbolTable::lookupNearestSymbolFrom<func::FuncOp>(op, op.getCalleeAttr());
+        if (!callee)
+            return signalPassFailure();
+
+        // Now let's create a new function to place the differentiation code into, so it doesn't
+        // pollute the current scope. We'll insert the new function after the callee.
+        {
+            // Insertion guards are useful to store the current IR position (insertion point) on stack,
+            // returning to it upon exiting the C++ scope.
+            PatternRewriter::InsertionGuard insertGuard(rewriter);
+            rewriter.setInsertionPointAfter(callee);
+
+            // Specify the properties of the new function like name, type signature, and visibility.
+            std::string fnName = op.getCallee().str() + ".finitediff";
+            StringAttr visibility = rewriter.getStringAttr("private");
+            // The function type should be identical to the type signature of the grad operation.
+            FunctionType fnType = rewriter.getFunctionType(op.getOperandTypes(), op.getResultTypes());
+
+            gradFn = rewriter.create<func::FuncOp>(op.getLoc(), fnName, fnType, visibility, nullptr, nullptr);
+
+            // Now we just to populate the actual body of the function. First create an empty body.
+            Block *fnBody = gradFn.addEntryBlock();
+            // Move the insertion point to inside the function body.
+            rewriter.setInsertionPointToStart(fnBody);
+            // Populate the function body.
+            populateFiniteDiffMethod(rewriter, op, gradFn);
+        }
+    }
+
+Symbols are string references to IR objects, which rather than containing a physical reference or
+pointer to the actual object, only refer to it by name. In order to dereference a symbol we always
+have to look it up in a symbol table. This means that symbols are a bit more flexible and don't
+have the same constraints as the `SSA <https://en.wikipedia.org/wiki/Static_single-assignment_form>`_
+values used everywhere else in the IR.
+
+To help visualize the process, after this step we would have gone from the IR shown above:
+
+.. code-block::
+
+    func.func @my_func(%x: f64, %y: f64, %z: f64) -> f64 {
+        ...
+    }
+
+    %deriv:3 = gradient.grad "fd" @my_func(%x, %y, %z) : (f64, f64, f64) -> (f64, f64, f64)
+
+to the following IR:
+
+.. code-block::
+
+    func.func @my_func(%x: f64, %y: f64, %z: f64) -> f64 {
+        ...
+    }
+
+    func.func @my_func.finitediff(f64, f64, f64) -> (f64, f64, f64) {
+        <contents of populateFiniteDiffMethod>
+    }
+
+    %deriv:3 = gradient.grad "fd" @my_func(%x, %y, %z) : (f64, f64, f64) -> (f64, f64, f64)
+
+Let's fill out the rest of the method. The finite-difference method works by invoking the callee
+twice with slightly different parameter values, approximating the partial derivative as follows:
+
+.. math::
+
+    \frac{d}{dx} f(x, y, z) \approx \frac{f(x + h, y, z) - f(x, y, z)}{h}
```

Comment:
It does not render well

Reply:
@rmoyard that's odd, it looks good to me: 
![image](https://github.com/PennyLaneAI/catalyst/assets/2959003/9353d8c4-b8a7-47c3-99e5-60eb0e737ea4)

Reply:
What do you see @rmoyard ?

Reply:
Yes I reopened and it looks good!